osu.Game/Screens/Play/SkipOverlay.cs

```diff
@@ -210,6 +212,17 @@ public void OnReleased(KeyBindingReleaseEvent<GlobalAction> e)
         {
         }
 
+        protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint, EffectControlPoint effectPoint, ChannelAmplitudes amplitudes)
+        {
+            base.OnNewBeat(beatIndex, timingPoint, effectPoint, amplitudes);
+
+            if (fadeOutBeginTime <= gameplayClock.CurrentTime)
+                return;
+
+            float progress = (float)(gameplayClock.CurrentTime - displayTime) / (float)(fadeOutBeginTime - displayTime);
```

Comment:
Why is `fadeOutBeginTime` checked against current time (therefore clamping from the left), but progress is not forcefully clamped to be no larger than 1?